### PR TITLE
libc: enable test on dummyfs targets

### PR DESCRIPTION
**Pull request by damianloew**
## Description

 - libc: enable on dummyfs targets
 - libc: temporary disable truncate/ftruncate up cases
 - libc: adjust setpuids_setsid case in unistd_uids.c to dummyfs targets
 - libc: create tmp and etc if didn't exist
 - libc: increase stack size

## Motivation and Context

 - JIRA: PD-329

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: libc/main.c [hunk 1]
```diff
@@ -15,12 +15,19 @@
...
 /* no need for forward declarations, RUN_TEST_GROUP does it by itself */
 void runner(void)
 {
 	setenv("POSIXLY_CORRECT", "y", 1);
 
+	/* the following files may not be present on dummyfd targets,
+	create them to make libc tests common */
+	create_dir_if_missing("/tmp");
+	create_dir_if_missing("/etc");
+	create_file_if_missing("/etc/passwd", "root:0B1ANiYi45IhxkfmUW155/GBd4IRE=:0:0:root:/:/bin/sh");
```
Comment (jsarzy) [inline libc/main.c:29]: Did you make sure that we can use `TEST_ASEERT*` here? It is allowed to call it in runner function or maybe only in `RUN_GROUP_TEST*`?

Comment (damianloew) [reply libc/main.c]: Yeah, it wasn't allowed. Thanks for this suggestion and please check the new approach

File: libc/main.c [hunk 2]
```diff
@@ -39,13 +39,68 @@ void runner(void)
...
+{
+	struct stat buffer;
 
-	unsetenv("POSIXLY_CORRECT");
+	if ((stat(path, &buffer) != 0) && (errno == ENOENT)) {
+		if (mkdir(path, S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH) != 0) {
+			fprintf(stderr, "Creating %s directory by mkdir failed: %s\n", path, strerror(errno));
+			return -1;
+		}
+	}
```
Comment (agkaminski) [inline libc/main.c:55]: What if `errno != ENOENT`?

File: libc/main.c [hunk 3]
```diff
@@ -39,13 +39,68 @@ void runner(void)
...
+static int libc_createFileIfMissing(const char *path, const char *charData)
+{
+	struct stat buffer;
+
+	if ((stat(path, &buffer) != 0) && (errno == ENOENT)) {
+		if (_create_file(path, charData) != 0) {
+			fprintf(stderr, "Creating %s file failed: %s\n", path, strerror(errno));
+			return -1;
+		}
+	}
```
Comment (agkaminski) [inline libc/main.c:71]: What if `errno != ENOENT`?

Comment (agkaminski) [inline libc/main.c:71]:
```suggestion
	if (stat(path, &buffer) != 0) {
		if (errno == ENOENT) {
			if (_create_file(path, charData) != 0) {
				fprintf(stderr, "Creating %s file failed: %s\n", path, strerror(errno));
				return -1;
			}
			else {
				return -1;
			}
		}
	}
```

File: libc/main.c [hunk 4]
```diff
@@ -39,13 +39,80 @@ void runner(void)
...
+
+	if (setenv(var, "y", 1) != 0) {
+		fprintf(stderr, "Setting %s environment variable failed: %s\n", var, strerror(errno));
+		return 1;
+	}
+
+	/* the following files may not be present on dummyfd targets,
+	create them to make libc tests common */
+	if (libc_createDirIfMissing("/tmp") < 0) {
+		return 1;
```
Comment (agkaminski) [inline libc/main.c:101]: Sorry, missed that on the previous review. If it is important to `unsetenv` before exit, we should do this on these error exit paths too

Comment (damianloew) [reply libc/main.c:101]: Yeah, but I think not so important to check its return code every time, so I would prefer sth like this:
```C
int main(int argc, char *argv[])
{
	const char *var = "POSIXLY_CORRECT";

	if (setenv(var, "y", 1) != 0) {
		fprintf(stderr, "Setting %s environment variable failed: %s\n", var, strerror(errno));
		return 1;
	}

	/* the following files may not be present on dummyfd targets,
	create them to make libc tests common */
	if (libc_createDirIfMissing("/tmp") < 0) {
		unsetenv(var);
		return 1;
	}
	if (libc_createDirIfMissing("/etc") < 0) {
		unsetenv(var);
		return 1;
	}
	if (libc_createFileIfMissing("/etc/passwd", "root:0B1ANiYi45IhxkfmUW155/GBd4IRE=:0:0:root:/:/bin/sh")) {
		unsetenv(var);
		return 1;
	}

	UnityMain(argc, (const char **)argv, runner);

	if (unsetenv(var) != 0) {
		fprintf(stderr, "Unsetting %s environment variable failed: %s\n", var, strerror(errno));
		return 1;
	}

	return 0;
}
```

**Review by agkaminski — APPROVED** [review]

File: libc/main.c [hunk 5]
```diff
@@ -39,13 +39,80 @@ void runner(void)
...
+			return -1;
+		}
+	}
+
+	return 0;
+}
+
+
+/* crete file with optional data (may be NULL) unless it exists */
+static int libc_createFileIfMissing(const char *path, const char *charData)
```
Comment (nalajcie) [inline libc/main.c:68]: IMHO naming functions which are not part of a libc - `libc_something` is really misleading

**Pull request merged**